pageserver/page_api/proto/page_service.proto

```diff
@@ -252,3 +258,17 @@ message GetSlruSegmentRequest {
 message GetSlruSegmentResponse {
   bytes segment = 1;
 }
+
+// Acquires or extends a lease on the given LSN. This guarantees that the Pageserver won't garbage
+// collect the LSN until the lease expires. Must be acquired on all relevant shards.
+message LeaseLsnRequest {
+  // The LSN to lease. Can't be 0 or below the current GC cutoff.
+  uint64 lsn = 1;
+}
+
+// Lease acquisition response. If the lease could not be granted because the LSN has already been
+// garbage collected, a FailedPrecondition status will be returned instead.
+message LeaseLsnResponse {
+  // The lease expiration time (exclusive; valid up to and including this timestamp).
```

Comment:
"valid up to and including this timestamp" sounds inclusive rather than exclusive.

It's probably not worth mentioning whether it's inclusive or exclusive anyway, because you can't really do anything with that information. A protobuf has nanosecond resolution, and it's nonsensical to even think about whether the lease expires one nanosecond earlier or later. You must renew the lease much much earlier anyway, because there's clock skew, network latencies etc.